pkg/storage/ingest/partition_offset_reader.go

```diff
@@ -120,10 +120,12 @@ func (p *partitionOffsetReader) getAndNotifyLastProducedOffset(ctx context.Conte
 	promise.notify(offset, err)
 }
 
-// getLastProducedOffset fetches and returns the last produced offset for a partition, or -1 if the
+// RequestLastProducedOffset fetches and returns the last produced offset for a partition, or -1 if the
 // partition is empty. This function issues a single request, but the Kafka client used under the
 // hood may retry a failed request until the retry timeout is hit.
-func (p *partitionOffsetReader) getLastProducedOffset(ctx context.Context) (_ int64, returnErr error) {
+//
+// This function may be directly called from outside the partitionOffsetReader.
+func (p *partitionOffsetReader) RequestLastProducedOffset(ctx context.Context) (_ int64, returnErr error) {
```

Comment:
at this point we have `FetchLastProducedOffset` and `RequestLastProducedOffset`. Can we clarify naming? Maybe turn them into something like `LazyLastProducedOffset` and `EagerLastProducedOffset` or `LastProducedOffset` and `ForceLastProducedOffset`?

Reply:
In https://github.com/grafana/mimir/pull/7486/commits/9f6340d7a23406455d79b46b1d637fbd774c4258 I've renamed the functions trying to follow what the exactly do:

- `FetchLastProducedOffset()`
- `WaitNextFetchLastProducedOffset()`